server/console_account.go

```diff
@@ -208,6 +256,188 @@ func (s *ConsoleServer) GetWalletLedger(ctx context.Context, in *console.Account
 	return &console.WalletLedgerList{Items: consoleLedger}, nil
 }
 
+func (s *ConsoleServer) ListAccounts(ctx context.Context, in *console.ListAccountsRequest) (*console.AccountList, error) {
+	// Searching only through tombstone records.
+	if in.Tombstones {
+		var userID *uuid.UUID
+		if in.Filter != "" {
+			uid, err := uuid.FromString(in.Filter)
+			if err != nil {
+				// Filtering for a tombstone using username, no results are possible.
+				return &console.AccountList{
+					TotalCount: countAccounts(ctx, s.logger, s.db),
+				}, nil
+			}
+			userID = &uid
+		}
+
+		if userID != nil {
+			// Looking up a single specific tombstone.
+			var createTime pgtype.Timestamptz
+			err := s.db.QueryRowContext(ctx, "SELECT create_time FROM user_tombstone WHERE user_id = $1", *userID).Scan(&createTime)
+			if err != nil {
+				if err == sql.ErrNoRows {
+					return &console.AccountList{
+						TotalCount: countAccounts(ctx, s.logger, s.db),
+					}, nil
+				}
+				s.logger.Error("Error looking up user tombstone.", zap.Any("in", in), zap.Error(err))
+				return nil, status.Error(codes.Internal, "An error occurred while trying to list users.")
+			}
+
+			return &console.AccountList{
+				Users: []*api.User{
+					{
+						Id:         in.Filter,
+						UpdateTime: &timestamp.Timestamp{Seconds: createTime.Time.Unix()},
+					},
+				},
+				TotalCount: countAccounts(ctx, s.logger, s.db),
+			}, nil
+		}
+
+		query := "SELECT user_id, create_time FROM user_tombstone LIMIT 50"
```

Comment:
Will this need to have pagination or we'll just return the first 50 results?

Reply:
Will need pagination. I didn't actually edit any of the Go code.